Source/File.System.Stub.UnitTests/EnumerableExtensionsTests.cs

```diff
@@ -0,0 +1,89 @@
+﻿using File.System.Stub.UnitTests.Helpers;
+using FluentAssertions;
+using NUnit.Framework;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace File.System.Stub.UnitTests
+{
+	[TestFixture]
+	internal static class EnumerableExtensionsTests
+	{
+		[Test]
+		public static void ItShouldReturnTrueWhenEnumerablesAreNull()
+		{
+			IEnumerable<SimpleObject> first = Enumerable.Empty<SimpleObject>();
+			IEnumerable<SimpleObject> second = Enumerable.Empty<SimpleObject>();
+
+			bool result = first.ElementsEqual(second, obj => obj.Key);
+
+			result.Should().BeTrue();
+		}
+
+		[Test]
+		public static void ItShouldReturnTrueWhenEnumerablesAreEmpty()
+		{
+			IEnumerable<SimpleObject> first = null;
+			IEnumerable<SimpleObject> second = null;
+
+			bool result = first.ElementsEqual(second, obj => obj.Key);
+
+			result.Should().BeTrue();
+		}
+
+		[Test]
+		public static void ItShouldReturnFalseWhenFirstEnumerableIsNull()

```

Comment:
Maybe we should treat empty and null collections equal? Something to think about.

Reply:
So empty = null ?

Reply:
Yes